Android/app/src/main/java/com/example/androidgui/useractivities/LoginActivity.java

```diff
@@ -54,7 +62,7 @@ public void runLogin(View view) {
             setEmptyErrorMessage(editTextPassword);
         } else {
             String id = editTextUserId.getText().toString();
-            this.controller.runLogin(id, editTextPassword.getText().toString());
+            this.loginPresenter.runLogin(id, editTextPassword.getText().toString());
             switch (this.loginResult) {
```

Comment:
I believe the presenter should be checking this and running the correct method if it fails?